test/make/TestFixDepsFile.gmk

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  Oracle designates this
+# particular file as subject to the "Classpath" exception as provided
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+default: all
+
+include $(SPEC)
+include MakeBase.gmk
+include UtilsForTests.gmk
+
+THIS_FILE := $(TOPDIR)/test/make/FixDepsFile.gmk
+DEPS := $(THIS_FILE) \
+    $(TOPDIR)/make/common/NativeCompilation.gmk \
+    #
+
+OUTPUT_DIR := $(TESTMAKE_OUTPUTDIR)/fix-deps-file
+$(call MakeDir, $(OUTPUT_DIR))
+
+################################################################################
+# The relevant case to test is when absolute paths aren't allowed.
+ALLOW_ABSOLUTE_PATHS_IN_OUTPUT := false
+FILE_MACRO_CFLAGS :=
+include NativeCompilation.gmk
+
+DEPS_FILE := $(OUTPUT_DIR)/deps.d
+
+test-fix-deps-file:
+	$(ECHO) "foo/bar1: \\" > $(DEPS_FILE).tmp
+	$(ECHO) "foo/baz1" >> $(DEPS_FILE).tmp
+	$(ECHO) "foo/bar : bar \\" >> $(DEPS_FILE).tmp
+	$(ECHO) " ./bar/baz /foo/baz" >> $(DEPS_FILE).tmp
+	$(call fix-deps-file, $(DEPS_FILE))
+	$(ECHO) "$(WORKSPACE_ROOT)/foo/bar1: \\" > $(DEPS_FILE).expected
+	$(ECHO) "$(WORKSPACE_ROOT)/foo/baz1" >> $(DEPS_FILE).expected
+	$(ECHO) "$(WORKSPACE_ROOT)/foo/bar : \\" >> $(DEPS_FILE).expected
+	$(ECHO) " $(WORKSPACE_ROOT)/bar \\" >> $(DEPS_FILE).expected
+	$(ECHO) " $(WORKSPACE_ROOT)/bar/baz \\" >> $(DEPS_FILE).expected
+	$(ECHO) " /foo/baz" >> $(DEPS_FILE).expected
+	$(DIFF) $(DEPS_FILE).expected $(DEPS_FILE)
```

Comment:
Does this need to be:
$(DIFF) $(DEPS_FILE).expected $(DEPS_FILE).tmp

Reply:
No, the fix-deps-file macro takes file.tmp as input and outputs into file, $(DEPS_FILE) is the output file from the macro in this case.

Reply:
OK